phpBB/includes/acp/acp_language.php

```diff
@@ -19,130 +19,211 @@
-					'lang_author'			=> $request->variable('lang_author', $row['lang_author'], true),
+					'lang_english_name'		=> $this->request->variable('lang_english_name', $row['lang_english_name']),
+					'lang_local_name'		=> $this->request->variable('lang_local_name', $row['lang_local_name'], true),
+					'lang_author'			=> $this->request->variable('lang_author', $row['lang_author'], true),
 				);
 
-				$db->sql_query('UPDATE ' . LANG_TABLE . '
-					SET ' . $db->sql_build_array('UPDATE', $sql_ary) . '
+				$this->db->sql_query('UPDATE ' . LANG_TABLE . '
+					SET ' . $this->db->sql_build_array('UPDATE', $sql_ary) . '
 					WHERE lang_id = ' . $lang_id);
 
-				$phpbb_log->add('admin', $user->data['user_id'], $user->ip, 'LOG_LANGUAGE_PACK_UPDATED', false, array($sql_ary['lang_english_name']));
+				$this->log->add('admin', $this->user->data['user_id'], $this->user->ip, 'LOG_LANGUAGE_PACK_UPDATED', false, array($sql_ary['lang_english_name']));
 
-				trigger_error($user->lang['LANGUAGE_DETAILS_UPDATED'] . adm_back_link($this->u_action));
+				trigger_error($this->language->lang('LANGUAGE_DETAILS_UPDATED') . adm_back_link($this->u_action));
 			break;
 
 			case 'details':
 
 				if (!$lang_id)
 				{
-					trigger_error($user->lang['NO_LANG_ID'] . adm_back_link($this->u_action), E_USER_WARNING);
+					trigger_error($this->language->lang('NO_LANG_ID') . adm_back_link($this->u_action), E_USER_WARNING);
 				}
 
 				$this->page_title = 'LANGUAGE_PACK_DETAILS';
 
 				$sql = 'SELECT *
 					FROM ' . LANG_TABLE . '
 					WHERE lang_id = ' . $lang_id;
-				$result = $db->sql_query($sql);
-				$lang_entries = $db->sql_fetchrow($result);
-				$db->sql_freeresult($result);
+				$result = $this->db->sql_query($sql);
+				$lang_entries = $this->db->sql_fetchrow($result);
+				$this->db->sql_freeresult($result);
 
 				if (!$lang_entries)
 				{
-					trigger_error($user->lang['LANGUAGE_PACK_NOT_EXIST'] . adm_back_link($this->u_action), E_USER_WARNING);
+					trigger_error($this->language->lang('LANGUAGE_PACK_NOT_EXIST') . adm_back_link($this->u_action), E_USER_WARNING);
 				}
 
 				$lang_iso = $lang_entries['lang_iso'];
 
-				$template->assign_vars(array(
-					'S_DETAILS'			=> true,
-					'U_ACTION'			=> $this->u_action . "&amp;action=details&amp;id=$lang_id",
-					'U_BACK'			=> $this->u_action,
-
-					'LANG_LOCAL_NAME'	=> $lang_entries['lang_local_name'],
-					'LANG_ENGLISH_NAME'	=> $lang_entries['lang_english_name'],
-					'LANG_ISO'			=> $lang_iso,
-					'LANG_AUTHOR'		=> $lang_entries['lang_author'],
-					'L_MISSING_FILES'			=> $user->lang('THOSE_MISSING_LANG_FILES', $lang_entries['lang_local_name']),
-					'L_MISSING_VARS_EXPLAIN'	=> $user->lang('THOSE_MISSING_LANG_VARIABLES', $lang_entries['lang_local_name']),
+				try
+				{
+					$lang_cfg = $this->language_helper->get_language_data_from_composer_file("{$this->phpbb_root_path}language/$lang_iso/composer.json");
```

Comment:
My preference generally is concatenation by `.`, but this rather a small thing, so feel free to ignore me.